index.bs

```diff
@@ -1520,94 +1459,189 @@ If the attestation root certificate is not dedicated to a single WebAuthn Authen
 should be specified in the attestation certificate itself, so that it can be verified against the <a>authenticatorData</a>.
 
 
+# [RP] Operations # {#rp-operations}
+
+Upon successful execution of a {{makeCredential()}} or {{getAssertion()}} call, the [RP]'s script receives a
+{{ScopedCredentialInfo}} or {{AuthenticationAssertion}} structure respectively from the client. It must then deliver the
+contents of this structure to the <a>[RP]</a>, using methods outside the scope of this specification. This section describes the
+operations that the [RP] must perform upon receipt of these structures.
+
+
+## Registering a new credential ## {#registering-a-new-credential}
+
+When requested to register a new credential with a {{ScopedCredentialInfo}} structure, a [RP] must proceed as follows:
+
+1. Perform JSON decoding on the {{ScopedCredentialInfo/clientData}} field of the {{ScopedCredentialInfo}} object to extract the
+    {{ClientData}} used for the credential's attestation.
+
+2. Verify that the {{ClientData/challenge}} in the {{ClientData}} matches the challenge that was sent to the authenticator in
+    the {{makeCredential()}} call.
+
+3. Verify that the {{ClientData/origin}} in the {{ClientData}} matches the [RP]'s origin.
+
+4. Verify that the {{ClientData/tokenBinding}} in the {{ClientData}} matches the token binding ID for the TLS connection over
+    which the attestation was obtained.
+
+5. Verify that the {{ClientData/extensions}} in the {{ClientData}} is a proper subset of the extensions requested by the RP.
+
+6. Compute the <a>clientDataHash</a> over {{ScopedCredentialInfo/clientData}} using the {{ClientData/hashAlg}} algorithm found
```

Comment:
See above. Will experiment with removing the signature format section in next commit.